core/base-service/base.js

```diff
@@ -106,9 +106,7 @@ module.exports = class BaseService {
    *  - capture: Array of names for the capture groups in the regular
    *             expression. The handler will be passed an object containing
    *             the matches.
-   *  - queryParams: Array of names for query parameters which will the service
-   *                 uses. For cache safety, only the whitelisted query
-   *                 parameters will be passed to the handler.
+   *  - queryParamSchema: Joi schema for valid query params.
```

Comment:
This certainly could be improved. There basically are two types, boolean params and string params, and we should explain each one. If anyone wants to take a stab at that, please do!

Reply:
As part of this PR or a follow up? My takeaway has been that for booleans we need use `Joi.equal('')` although I'm still not sure how to best articulate _why_ that's the case.

IMO it'd be sufficient to just mention that fact for booleans in the comment doc (and probably the tutorial at some point too). Or do you think explanation/details accompany it in the comments?

Reply:
I opened #3170 which adds a usage comment. It partly answers why… see if that seems complete enough an explanation?

I don't really like how the query params are handled by the badge builder; seems like it could be time for another redesign of the builder. I'll open an issue with some proposals.